wormhole-connect/src/store/transferInput.ts

```diff
@@ -230,16 +245,27 @@ export const transferInputSlice = createSlice({
     },
     setBalances: (
       state: TransferInputState,
-      { payload }: PayloadAction<{ chain: ChainName; balances: Balances }>,
+      {
+        payload,
+      }: PayloadAction<{
+        address: WalletAddress;
+        chain: ChainName;
+        balances: Balances;
+      }>,
     ) => {
-      const { chain, balances } = payload;
-      state.balances = {
-        ...state.balances,
-        ...{
-          [chain]: {
-            lastUpdated: Date.now(),
-            balances: { ...state.balances[chain], ...balances },
-          },
+      const { chain, balances, address } = payload;
+      if (!address) return;
+      // create new field for address
+      if (!state.balances[address]) {
+        state.balances = {
+          ...state.balances,
+          ...{ [address]: { [chain]: {} } },
+        };
+      }
+      state.balances[address] = {
+        [chain]: {
```

Comment:
I think it's missing an object destructuring. When setting for a new chain it'll clear the other existing chains.

```suggestion
        ...state.balances[address],
        [chain]: {
```